lib/src/builder/generation/impl_generation.dart

```diff
@@ -247,8 +247,9 @@ class ImplGenerator {
             ..writeln()
             ..writeln('  @override')
             ..writeln('  set state(Map value) {')
-            ..writeln('    super.state = value;')
-            ..writeln('    _cachedTypedState = typedStateFactoryJs(value);')
+            ..writeln('    JsBackedMap jsBackedValue = JsBackedMap.from(value);')
```

Comment:
This is creating a copy of the state map, which we want to avoid when possible:
- for performance (one of the reasons Component2 is faster is fewer map copies).
- to ensure the state value is in sync with `this.state`


Sorry, we should have groomed this ticket before working on it. I think instead of making this setter suboptimal, we should drop support for initializing state via `this.state = ` and instead create a method in Component2 like `initalizeState(Map value)` or something that should be used instead. Thoughts?

If we do go that route, I think it would be nice here to add an assert that it is indeed a JsBackedMap in case someone tries to do it the React JS way anyways.
```dart
assert(value is JsBackedMap, 
    'Component2.state may only be set to a JsBackedMap, '
        'and must not be set outside of react-dart internals.')
```

Reply:
Oh that makes sense. Sorry about that - I feel like I should have seen that one. 

For avoiding `this.state =`, we already overrode the `[]=` operator to discourage setting state / props directly in https://github.com/Workiva/over_react/pull/249, so that solution makes sense and seems consistent. Implementation wise, it would be implementing the logic in react-dart and then exposing in in over_react, right @greglittlefield-wf ?